### PR TITLE
Align tooltip drawing method with Minecrafts method

### PR DESCRIPTION
**Pull request by davenonymous**
The method used by vanilla can be found in the
net.minecraftforge.fml.client.config.GuiUtils class and it very
closely resembles the one used in the AntiqueAtlas GuiComponent
class.

I've cleaned up this version to use the same deobfuscated names
as the Minecraft one and added the firing of the Events currently
being fired by forge.

This allows mods like Colored Tooltips to modify the background
color of the Antique Atlas tooltips.

Ping @Darkosto

### CODE REVIEW

**Comment by TehNut** [comment]
Is there a reason it can't just be moved to use the Forge method?

**Comment by davenonymous** [comment]
It is slightly modified to work together with other gui components, e.g. it disables the previously set stencil bit - but it might be enough to just wrap this around the original method. I'll take another look.
Edit: It also has a slightly different positioning calculation. Using the forge method the tooltip is being positioned wrongly.

File: src/main/java/hunternif/mc/atlas/client/gui/core/GuiComponent.java [hunk 1]
```diff
@@ -438,69 +441,90 @@ protected boolean isMouseInRadius(int x, int y, int radius) {
...
 				}
 
-				j1 += 10;
+				tooltipY += 10;
 			}
+
+			MinecraftForge.EVENT_BUS.post(new RenderTooltipEvent.PostText(ItemStack.EMPTY, textLines, tooltipX, tooltipTop, font, tooltipTextWidth, tooltipHeight));
+
 			if (stencilEnabled) GL11.glEnable(GL11.GL_STENCIL_TEST);
+			GlStateManager.enableLighting();
```
Comment (tyra314) [inline src/main/java/hunternif/mc/atlas/client/gui/core/GuiComponent.java:524]: I'm not too familiar with OpenGL, but I'd expect the exact reverse ordering of these calls when compared to the disable calls at the beginning of this function?

**Review by tyra314 — COMMENTED** [review]
Have you actually tested, if the tooltip can be changed by other mods with this patchset?

**Comment by davenonymous** [comment]
I think this PR is out of date, since some of the internal logic regarding tooltips in forge has changed.
See https://github.com/MinecraftForge/MinecraftForge/pull/4595

All in all you should indeed rather switch to calling the forge methods, so you don't have to deal with this.

**Pull request closed without merging**